dot/state/storage.go

```diff
@@ -305,3 +308,27 @@ func (s *StorageState) GenerateTrieProof(stateRoot common.Hash, keys [][]byte) (
 	encodedProofNodes [][]byte, err error) {
 	return proof.Generate(stateRoot[:], keys, s.db)
 }
+
+// GetRuntime gets the runtime for the corresponding block hash from storageState
+func (s *StorageState) GetRuntime(blockHash common.Hash) (instance runtime.Instance, err error) {
```

Comment:
why are we adding a `GetRuntime` method to the storage state? Shouldn't the `blockTree` contain all the runtimes?

Reply:
This PR introduces clearRuntimes that removes all but the current runtime from blockTree, so we need a way to get runtimes from storage state.